internal/bminventory/inventory.go

```diff
@@ -919,6 +919,25 @@ func (b *bareMetalInventory) DeregisterClusterInternal(ctx context.Context, para
 		log.Warnf("failed to delete DNS record sets for base domain: %s", cluster.BaseDNSDomain)
 	}
 
+	for _, h := range cluster.Hosts {
+		infraEnv, err2 := common.GetInfraEnvFromDB(b.db, h.InfraEnvID)
+		if err2 != nil {
+			log.WithError(err2).Errorf("failed to get infra env: %s", h.InfraEnvID)
+			return common.NewApiError(http.StatusInternalServerError, err2)
+		}
+		if infraEnv.ClusterID == *cluster.ID {
+			if err = b.hostApi.UnRegisterHost(ctx, h.ID.String(), h.InfraEnvID.String()); err != nil {
+				log.WithError(err).Errorf("failed to delete host: %s", h.ID.String())
+				return common.NewApiError(http.StatusInternalServerError, err)
```

Comment:
Is it always internal error?

Reply:
`UnRegisterHost`  returns the DB error.
Should we check for more granularity?

Reply:
Maybe let the api  level to handle the error type if it's 500?

Reply:
I update the code. See if it is what you meant.